packages/engine/src/scene/components/SplineComponent.ts

```diff
@@ -23,33 +23,230 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
-import { Vector3 } from 'three'
+import {
+  ArrowHelper,
+  BoxGeometry,
+  BufferAttribute,
+  BufferGeometry,
+  CatmullRomCurve3,
+  Euler,
+  Group,
+  Line,
+  LineBasicMaterial,
+  Mesh,
+  MeshLambertMaterial,
+  Quaternion,
+  Vector3
+} from 'three'
 
+import { matches } from '../../common/functions/MatchesUtils'
 import { defineComponent } from '../../ecs/functions/ComponentFunctions'
+import { ObjectLayers } from '../constants/ObjectLayers'
+import { setObjectLayers } from '../functions/setObjectLayers'
+import { addObjectToGroup, removeObjectFromGroup } from './GroupComponent'
 
-export type SplineComponentType = {
-  splinePositions: Vector3[]
+// an idea of a bundle that packages up all the related concerns on a spline point
+// @todo there may need to be a concept of an engine mode versus editor mode where no gizmos are manufactured
```

Comment:
we are actually wanting to move away from this dileniation, in order to bring the studio and runtime location closer together, such that we can use the editor functionality in immersive sessions, and to reduce complexity. gizmos should only be created when the gizmo tool is enabled.